.github/actions/integration-tests/prepare-run-file

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Helper script to prepare run-file json
+# Usage:
+#     source ${SCRIPT_DIR}/prepare-run-file
+
+function replace_value() {
+    # Replace a value in the run-file json
+    # Usage: replace_value remote_host ${remote_host} \
+    #                      '."endpoints"[0]."host" = $remote_host'
+    # ("host" key will be replaced with the value of $remote_host variable)
+    #
+    # --arg <variable> <value> is declaring an argument named <variable>,
+    # with a value of <value> in the jq environment
+    # The <query> is what specifies the location in the JSON
+    # (the first part) and then the value to set it to (the second part, after the =)
+
+    local json=$1; shift # json file to replace value
+    local variable=$1; shift # variable to be declared as variable=value in jq's env
+    local value=$1; shift # value to assign to the variable declaration in jq's env
+    local query=$1; shift # query to match json's key/val
+
+    if [ -z "$(command -v jq)" ]; then
+        echo "ERROR running jq to edit the run-file JSON."
+        exit 1
+    fi
+
+    jq --arg ${variable} "${value}" "${query}" ${json} > ${json}.tmp
+    if [ $? != 0 ]; then
+        echo "ERROR editing the run-file JSON."
+        exit 1
+    fi
+
+    # Update original run-file json
+    /bin/mv ${json}.tmp ${json}
+
+    # Remove temp json
+    /bin/rm ${json}.tmp
```

Comment:
The above command is a `mv` so you only need this in case of an error...which should probably be handled rather than blindly ignored if you think that is going to happen.

Reply:
oops